### DISCUSSION TITLE
Custom disconnect button reconnects on refresh

### DISCUSSION THREAD

**Discussion by madhawk1296** [rainbow-me/rainbowkit#890] [Q&A]
I'm using a custom disconnect button with wagmi's `useDisconnect` hook. It disconnects when I press the button, but whenever I refresh the page, it reconnects automatically. 

Is there anyway to fix this? thanks!

**Comment by aalmada** [comment]
I think I've never seen a connect button the really disconnects. Try the connect button at https://www.rainbowkit.com/ and you'll see the same behaviour.

I've previously discussed this at https://github.com/rainbow-me/rainbowkit/discussions/693#discussioncomment-4225740

**Reply by madhawk1296** [reply]
Yeah, I noticed that as-well. Every other app I've seen that integrates rainbowkit disconnects properly, they might be adding their own logic with `localStorage`.

**Reply by aalmada** [reply]
https://wagmi.sh/react/connectors/metaMask#shimdisconnect
